src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,64 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int SIZE_ARRAY = 10;
+    private final Object[] keyStorage;
+    private final Object[] valueStorage;
+    private int countIndex;
+
+    public StorageImpl(int size) {
+        keyStorage = new Object[size];
+        valueStorage = new Object[size];
+        countIndex = 0;
+    }
+
+    public StorageImpl() {
+        this(SIZE_ARRAY);
+    }
+
     @Override
     public void put(K key, V value) {
+        int indexKey = getIndexAndCheckExistKey(key);
+        if (indexKey >= 0) {
+            valueStorage[indexKey] = value;
+        } else {
+            keyStorage[countIndex] = key;
+            valueStorage[countIndex++] = value;
+
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int indexKey = getIndexAndCheckExistKey(key);
+        if (indexKey >= 0) {
+            return (V) valueStorage[indexKey];
+        } else {
+            return null;
+        }
     }
 
     @Override
     public int size() {
+        return countIndex;
+    }
+
+    private int getIndexAndCheckExistKey(K key) {
```

Comment:
Do not use 'and' in the method name. Never. It is against Single Responsibility Principle
```suggestion
    private int getKeyIndex(K key) {
```